### DISCUSSION TITLE
Set signal throttle/debounce

### DISCUSSION THREAD

**Discussion by FallingSnow** [solidjs/solid#923] [Ideas]
I feel like debouncing or throttling things like user input are a common enough occurrence that it might warrant adding these features as part of the options in `createSignal`. Although adding this "unnecessary" code would add technical debt and might have very very slight performance implications.

These options (`throttle`,`debounce`) could take a function or a number of milliseconds.

What are your thoughts on adding optional debounce/throttle functions to the signal setter?

**Comment by atk** [comment]
We usually prefer composability over integrated solutions. Have a look at the external solid-primitives packages for debounce and throttle.

**Reply by FallingSnow** [reply]
Oh, that's pretty cool. Didn't even know about `solid-primatives`.

**Reply by atk** [reply]
We aim to provide a collection of high-quality primitives that puts other frameworks to shame. 😁

If you find anything missing or want to contribute yourself, we have a channel ('#solid-primitives') on the solid discord: https://discord.com/invite/solidjs
